mongomock/filtering.py

```diff
@@ -401,11 +405,24 @@ def operator_eq(doc_val, search_val):
     return operator.eq(doc_val, search_val)
 
 
+def cmp_operator(op, a, b):
```

Comment:
Please treat this as a special case in the bson_compare function. Or did I miss something?

Reply:
this only apply to sorting operators. appart from checkimg wether an operator is one of the sorting operator in bson_compare, i'm not sure how to do it, and the former method does not seem better to me than having the correct operators right offt the bat.

Reply:
I'm not sure to understand what you're implying. `bson_compare` is only ever called with values from the `SORTING_OPERATOR_MAP` dict.

I think that comparing two bytes fields even when not sorting should use your method. `{'$lt': [b'abc', b'b']}`.

I suggest you keep the tests you have added and try moving the code to bson_compare: they should pass and not break any other tests.

Reply:
my bad dunno why, i though it was called for other operators. done.